src/CommonLib/aspnetcoreconfig.h

```diff
@@ -215,18 +215,38 @@ class ASPNETCORE_CONFIG : IHttpStoredContext
         return &m_struConfigPath;
     }
 
-    STRU*
-    QueryHostfxrPath()
+    PCWSTR*
+    QueryHostFxrArguments(
+        VOID
+    )
     {
-        return &m_struHostFxrPath;
+        return m_ppStrArguments;
     }
 
-    BOOL
-    QueryIsStandAloneApplication(
+    DWORD
+    QueryHostFxrArgCount(
+        VOID
+    )
+    {
+        return m_dwArgc;
+    }
+
+    STRU*
+    QueryHostFxrFullPath(
         VOID
     )
     {
-        return m_fIsStandAloneApplication;
+        return &m_struHostFxrLocation;
```

Comment:
the return type can be changed tp const PCWSTR i.e, return m_struHostFxrLocation->QueryStr(). This can prevent others from accidently change the stru.

Reply:
Currently the STRU is modified outside of config. Other pieces of config use this pattern where query returns a pointer to a STRU: https://github.com/aspnet/AspNetCoreModule/blob/dev/src/CommonLib/aspnetcoreconfig.h#L122-L128. I can make this a set/query pattern if you would like.